sds_in_a_box/sds_in_a_box_stack.py

```diff
@@ -85,20 +97,48 @@ def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
                                           code=lambda_.Code.from_asset(os.path.join(os.path.dirname(os.path.realpath(__file__)), "SDSCode")),
                                           handler="indexer.lambda_handler",
                                           role=lambda_role,
-                                          runtime=lambda_.Runtime.PYTHON_3_9,
+                                          runtime=lambda_.Runtime.PYTHON_3_7,
                                           timeout=cdk.Duration.minutes(15),
                                           memory_size=1000,
                                           environment={
                                             "OS_ADMIN_USERNAME": "master-user", 
-                                            "OS_ADMIN_PASSWORD_LOCATION": os_secret.secret_name,
+                                            "OS_ADMIN_PASSWORD_LOCATION": os_secret.secret_value.unsafe_unwrap(),
                                             "OS_DOMAIN": domain.domain_endpoint,
                                             "OS_PORT": "443",
-                                            "OS_INDEX": "metadata"
-                                            }
+                                            "OS_INDEX": "metadata",
+                                            "S3_BUCKET": data_bucket.s3_url_for_object()},
+                                            layers=[lambda_alpha_.PythonLayerVersion(self, "SDSIndexerCodeLayer", entry=os.path.join(os.path.dirname(os.path.realpath(__file__)), "SDSCode/"))]
```

Comment:
Rather than a layer here, can you actually make the entire Function through lambda_alpha? That should automatically create the appropriate layer for you I _think_.
https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_lambda_python_alpha/README.html#python-function

I think that might help with the Python Version required here too.

Reply:
This fixed the python version issue. Thanks!

Reply:
nice!